stellargraph/layer/graphsage.py

```diff
@@ -837,16 +849,15 @@ def apply_layer(x: List, num_hops: int):
             )
 
         # Form GraphSAGE layers iteratively
-        self.layer_tensors = []
         h_layer = xin
         for layer in range(0, self.max_hops):
             h_layer = apply_layer(h_layer, layer)
-            self.layer_tensors.append(h_layer)
 
         # Remove neighbourhood dimension from output tensors of the stack
         # note that at this point h_layer contains the output tensor of the top (last applied) layer of the stack
         h_layer = [
-            Reshape(K.int_shape(x)[2:])(x) for x in h_layer if K.int_shape(x)[1] == 1
+            Reshape(K.int_shape(x)[2:])(x) if K.int_shape(x)[1] == 1 else x
```

Comment:
Good catch!

Reply:
Only, I have no idea what mess it might make if the first dimension is ever not one!

Reply:
Well, it would have made a total mess before. Now, there's a chance of it working 😆

Reply:
Good catch indeed. Why did it work before?

Reply:
Because the code is designed to aggregate neighbourhoods down to a single feature vector per node, with leading dimension of 1. So I guess it always managed to do this - not sure in what cases it might possibly fail.

Reply:
Should this be generalised to
```
h_layer = [
            Reshape(K.int_shape(x)[2:])(x) if K.int_shape(x)[1] == head_shape else x
            for x in h_layer
        ]
```
?

Reply:
Possibly. Personally, I'd like to first see a use case where node features are not vectors.